ts/db.go

```diff
@@ -34,29 +38,106 @@ func NewDB(kv *client.KV) *DB {
 	}
 }
 
-// storeData attempts to store the supplied time series data on the server.
-// Data will be sampled at the supplied resolution.
-func (db *DB) storeData(r Resolution, data proto.TimeSeriesData) error {
-	internalData, err := data.ToInternal(r.KeyDuration(), r.SampleDuration())
-	if err != nil {
-		return err
+// PollSource begins a Goroutine which periodically queries the supplied
+// DataSource for time series data, storing the returned data in the server.
+// Stored data will be sampled using the provided Resolution. The polling
+// process will continue until the provided util.Stopper is stopped.
+func (db *DB) PollSource(source DataSource, frequency time.Duration, r Resolution, stopper *util.Stopper) {
+	p := &poller{
+		db:        db,
+		source:    source,
+		frequency: frequency,
+		r:         r,
+		stopper:   stopper,
+	}
+	p.start()
```

Comment:
Oh I see...this just creates and starts it doesn't return... Hmmm new pattern for the code here. That's fine. Ignore previous comment.

But I still think you should rename the method with a verb. How about:

`StartPollSource` ?